machine/v1beta1/types_vsphereprovider.go

```diff
@@ -70,6 +70,13 @@ type VSphereMachineProviderSpec struct {
 	// When using LinkedClone, if no snapshots exist for the source template, falls back to FullClone.
 	// +optional
 	CloneMode CloneMode `json:"cloneMode,omitempty"`
+	// Disks is a list of non OS disks to be attached to the VM.
```

Comment:
Godoc should explain validations, and should start with the serialised form (`disks`) of the field name.

What does a user need to do for each disk here, do they need to specify any software side configuration?

Reply:
Currently the logic is to just create the disk and attach to vsphere VMs.  What user does after that can be anything.  They can use LSO (local storage operator) or the future auto mount that we are currently working on to put these disks into certain locations.